add project readme (#11)
AfricasVoices/Project-UNICEF-COVID19-SOM: README.md

# Project-UNICEF-COVID19-SOM
Data pipeline for Project-UNICEF-COVID19-SOM.

This pipeline fetches all project data from a Rapid Pro instance and Coda, and processes it to produce data for
manual labelling and verification in Coda, CSV files suitable for analysis, and some automatic analysis of that data.

## Pre-requisites
Before the pipeline can be run, the following tools must be installed:
 - Docker
 - Bash
 
Development requires the following additional tools:
 - Python 3.6
 - pipenv

>  - git

Reviewer: Add a section that explains that you now need to pipenv --three sync after cloning

Author: updated thanks.

Author: @as2388  Please take another quick look.

## Usage

> A pipeline run consists of the following five steps, executed in sequence:

Reviewer: ```suggestion
A pipeline run consists of the following steps, executed in sequence:
```

Author: updated thanks.

1. Download coded data from Coda.
2. Fetch all the relevant data from Rapid Pro.
3. Process the raw data to produce the outputs required for coding and then for analysis.
4. Upload the new data to Coda for manual verification and coding.
5. Run automated analysis.
6. Back-up the project data root.

> 7. Upload execution logs/archives to long term storage.

Reviewer: Update to match the new run scripts

Author: updated thanks.

# Project-UNICEF-COVID19-SOM
Data pipeline for Project-UNICEF-COVID19-SOM.

This pipeline fetches all project data from a Rapid Pro instance and Coda, and processes it to produce data for
manual labelling and verification in Coda, CSV files suitable for analysis, and some automatic analysis of that data.

## Pre-requisites
Before the pipeline can be run, the following tools must be installed:
 - Docker
 - Bash
 
Development requires the following additional tools:
 - Python 3.6
 - pipenv
 - git
 

> After cloning you need to run `pipenv sync` in the project directory to install all dependencies required by the last step 

Reviewer: ```suggestion
After cloning run `$ pipenv sync` in the project directory to install all dependencies required by the last step 
```

participation maps. To use, run the following command from 
the `run_scripts` directory:

```
$ ./5_automated_analysis.sh <user> <pipeline-configuration-file-path> <data-root>
```

where:
- `user` is the identifier of the person running the script, for use in the TracedData Metadata 
  e.g. `user@africasvoices.org`.
- `pipeline-configuration-file-path` is an absolute path to a pipeline configuration json file.
- `data-root` is an absolute path to the directory in which all pipeline data should be stored.
  All output files will be saved in `<data-root>/Automated Analysis`.

### 6. Back-up the Data Directory
This stage makes a backup of the project data directory by creating a compressed, versioned, time-stamped copy at the
requested location.
To use, run the following command from the `run_scripts` directory:

```
$ ./6_backup_data_root.sh <data-root> <data-backups-dir>
```

where:
- `data-root` is an absolute path to the directory to back-up.
- `data-backups-dir` is a directory which the `data-root` directory will be backed-up to.
  The data is gzipped and given the name `data-<utc-date-time-now>-<git-HEAD-hash>`.

### 7. Upload analysis files

> This stage uploads the analysis files generated by generate outputs & automated analysis stage to Google Cloud storage.

Reviewer: ```suggestion
This stage uploads the analysis files generated by generate outputs & automated analysis stage to Google Drive.
```

To use, run the following command from the `run_scripts` directory:

```
$ ./7_upload_analysis_files.sh <user> <google-cloud-credentials-file-path> pipeline-configuration-file-path> <run-id> <data-root>

```

where:
- `user` is the identifier of the person running the script, for use in the TracedData Metadata 
  e.g. `user@africasvoices.org`.
- `google-cloud-credentials-file-path` is an absolute path to a json file containing the private key credentials
  for accessing a cloud storage credentials bucket containing all the other project credentials files.
- `pipeline-configuration-file-path` is an absolute path to a pipeline configuration json file.
- `run-id` is a unique identifier for the run being uploaded. This will be included in all of the uploaded file names.
- `data-root` is an absolute path to the directory in which all pipeline data should be stored.

> ### 8. Upload logs files

Reviewer: ```suggestion
### 8. Upload log files
```

This stage uploads the archive produced in the backup-data stage and a memory profile log of the generate outputs stage

> to project google drive directory.

Reviewer: ```suggestion
to google cloud storage.
```
